internal/command/test_test.go

```diff
@@ -3497,67 +4862,257 @@ func TestTest_InvalidOverrides(t *testing.T) {
 	code := c.Run([]string{"-no-color"})
 	output := done(t)
 
+	// ASSERTIONS
 	if code != 0 {
 		t.Errorf("expected status code 0 but got %d", code)
 	}
+	stdErr := output.Stderr()
+	if len(stdErr) > 0 {
+		t.Fatalf("unexpected error output:\n%s", stdErr)
+	}
 
-	expected := `main.tftest.hcl... in progress
-  run "setup"... pass
+	// State is stored according to the backend block
+	actualState := testStateRead(t, localStatePath)
+	if diff := cmp.Diff(actualState.String(), expectedState); len(diff) > 0 {
+		t.Fatalf("state didn't match expected:\nexpected:\n%s\nactual:\n%s\ndiff:\n%s", expectedState, actualState, diff)
+	}
 
-Warning: Invalid override target
+	if provider.ResourceCount() != 1 {
+		t.Fatalf("should have deleted all resources on completion except test_resource.a. Instead state contained %b resources: %v", provider.ResourceCount(), provider.ResourceString())
+	}
 
-  on main.tftest.hcl line 39, in run "setup":
-  39:     target = test_resource.absent_five
+	val := provider.Store.Get(resourceId)
 
-The override target test_resource.absent_five does not exist within the
-configuration under test. This could indicate a typo in the target address or
-an unnecessary override.
+	// If the ID hasn't changed then we've used the pre-existing state, instead of remaking the resource
+	if val.GetAttr("id").AsString() != resourceId {
+		t.Errorf("expected resource to have value %q but got %s", resourceId, val.GetAttr("id").AsString())
+	}
+	if val.GetAttr("value").AsString() != "value-from-run-that-controls-backend" {
+		t.Errorf("expected resource to have value 'value-from-run-that-controls-backend' but got %s", val.GetAttr("value").AsString())
+	}
+}
 
-  run "test"... pass
+// Testing whether a state artifact is made for a run block with a backend or not
+//
+// Artifacts are made when the cleanup operation errors.
+func TestTest_UseOfBackends_whenStateArtifactsAreMade(t *testing.T) {
```

Comment:
I observed that during in test case # 1, where no artefacts are made, a non-empty manifest file is created. When I debug the test and check the temp dir's files, the `.terraform/test/` directory only contains manifest.json (no state files) and the manifest refers to a non-existent file:

```json
{
  "version": 0,
  "files": {
    "main.tftest.hcl": {
      "states": {
        "": {
          "path": ".terraform/test/Qwlngn0M.tfstate",
          "reason": ""
        }
      }
    }
  }
}
```

Does this need to be fixed? This could be confusing when debugging issues with the test command, though arguably you could say that this is an implementation detail and users shouldn't be inspecting the file.